File: app/src/main/java/com/example/todoapp/LoginActivity.java
package com.example.todoapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {
    EditText full_name,user_name;
    Button login_button;
    SharedPreferences sharedPreferences;
    SharedPreferences.Editor editor;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        full_name=findViewById(R.id.full_name);
        user_name=findViewById(R.id.user_name);
        login_button=findViewById(R.id.login_button);
        setUpSharedPreferences();
        login_button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //Log.d("LoginActivity","button clicked");
                String FullName=full_name.getText().toString();
                String UserName=user_name.getText().toString();

                if(!FullName.isEmpty() && !UserName.isEmpty()){
                    Intent intent =new Intent(LoginActivity.this,MyNotesActivity.class);
                    intent.putExtra(AppConstant.FULL_NAME,FullName);
                    intent.putExtra(AppConstant.USER_NAME,UserName);
                    startActivity(intent);
                    //Log in status
                    saveLoginStatus();
                    saveUserName(UserName);

                }
                else{
                    Toast.makeText(LoginActivity.this, "You need to put your name buddy, don't forget your user name too", Toast.LENGTH_SHORT).show();
                }


            }
        });
    }

    private void saveUserName(String userName) {
        editor=sharedPreferences.edit();
        editor.putString(Shared_pref_constant.USER_NAME,userName);
        editor.apply();
    }

    private void saveLoginStatus() {
        editor=sharedPreferences.edit();
        editor.putBoolean(Shared_pref_constant.IS_LOGGED_IN,true);
        editor.apply();
    }

    private void setUpSharedPreferences() {
        sharedPreferences=getSharedPreferences(Shared_pref_constant.SHARED_PREF,MODE_PRIVATE);
    }
}

File: app/src/main/java/com/example/todoapp/View/LoginActivity.kt
package com.example.todoapp.View

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.todoapp.utils.AppConstant.FULL_NAME
import com.example.todoapp.utils.AppConstant.USER_NAME
import com.example.todoapp.R
import com.example.todoapp.utils.Shared_pref_constant
import com.example.todoapp.utils.Shared_pref_constant.IS_LOGGED_IN
import com.example.todoapp.utils.Shared_pref_constant.SHARED_PREF

class LoginActivity: AppCompatActivity() {
    lateinit var full_name: EditText
    lateinit var user_name: EditText
    lateinit var login_button: Button
    lateinit var sharedPreferences: SharedPreferences
    lateinit var editor: SharedPreferences.Editor


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        bindView()
        setUpSharedPreferences()

    }

    private fun setUpSharedPreferences() {
        sharedPreferences = getSharedPreferences(SHARED_PREF, Context.MODE_PRIVATE)
    }

    private fun bindView() {
        full_name = findViewById(R.id.full_name)
        user_name = findViewById(R.id.user_name)
        login_button = findViewById(R.id.login_button)
        login_button.setOnClickListener {
            //Log.d("LoginActivity","button clicked");
            val FullName = full_name.text.toString()
            val UserName = user_name.text.toString()
            if (!FullName.isEmpty() && !UserName.isEmpty()) {
                val intent = Intent(this@LoginActivity, MyNotesActivity::class.java)
                intent.putExtra(FULL_NAME, FullName)
                intent.putExtra(USER_NAME, UserName)
                startActivity(intent)
                //Log in status
                saveLoginStatus()
                saveUserName(UserName)
            } else {
                Toast.makeText(this@LoginActivity, "You need to put your name buddy, don't forget your user name too", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun saveUserName(userName: String) {
        editor = sharedPreferences.edit()
        editor.putString(Shared_pref_constant.USERNAME, userName)
        editor.apply()
    }

    private fun saveLoginStatus() {
        editor = sharedPreferences.edit()
        editor.putBoolean(IS_LOGGED_IN, true)
        editor.apply()
    }
}